src/notes.c

```diff
@@ -497,6 +497,44 @@ int git_note_create(
 	return error;
 }
 
+int git_note_commit_create(
+	git_oid *notes_commit_out,
+	git_oid *notes_blob_out,
+	git_repository *repo,
+	git_commit *parent,
+	const git_signature *author,
+	const git_signature *committer,
+	const git_oid *oid,
+	const char *note,
+	int allow_note_overwrite)
+{
+	int error;
+	git_tree *tree = NULL;
+	git_oid note_commit_oid, note_blob_oid;
+	char target[GIT_OID_HEXSZ + 1];
+
+	git_oid_tostr(target, sizeof(target), oid);
+
+	if (parent != NULL && (error = git_commit_tree(&tree, parent)) < 0)
+		goto cleanup;
+
+	error = note_write(&note_commit_oid, &note_blob_oid, repo, author, committer, NULL,
```

Comment:
Wait a second, this is strange. You're passing in `git_oid **`'s here, but `note_write` accepts `git_oid *` only.

Reply:
Ah, never mind. I've been confused by the naming here. You could've istead just passed in the pointers, but this shall not put your PR on hold now.

Reply:
Ah I see, so I'm duplicating the `git_oid_cpy`, yes that's a shame, I'll change it.